transforms/geocable.py

```diff
@@ -1102,61 +1245,80 @@ async def compile_rope(
             if node.config.coll_side_count >= 3:
                 has_coll = True
 
+        # All the configs should be the same, so just use the last node in the set.
+        conf = node.config
+
+        # If separate models are enabled, call get_model twice to get models for the frame and
+        # glass individually.
+        is_sep = conf.vac_separate_glass == VactubeGenType.SEPARATE
+        # First do the frame, or everything if we're not separating them.
         model_name, (light_origin, coll_data, seg_props, vac_points) = await compiler.get_model(
-            (frozenset(local_nodes), frozenset(connections)),
+            (frozenset(local_nodes), frozenset(connections), VactubeGenPartType.FRAME if is_sep else VactubeGenPartType.ALL),
             build_rope,
             (center, ctx.pack.fsys),
         )
+        modellist = [ModelContainer(model_name, light_origin, coll_data, seg_props, StaticPropFlags.NONE)]
+
+        if is_sep:
+            # Generate the glass only
+            model_name, (light_origin, coll_data, seg_props, _) = await compiler.get_model(
+                (frozenset(local_nodes), frozenset(connections), VactubeGenPartType.GLASS),
+                build_rope,
+                (center, ctx.pack.fsys),
+            )
+            modellist.append(ModelContainer(model_name, light_origin, coll_data, seg_props, StaticPropFlags.NO_SHADOW))
 
         if vac_points and vac_node_mod is not None:
             for track in vac_points:
                 vac_node_mod.SPLINES.append(vac_node_mod.Spline(center, track))
 
         # Compute the flags. Just pick a random node, from above.
-        conf = node.config
         flags = StaticPropFlags.NONE
         if conf.prop_light_bounce:
             flags |= StaticPropFlags.BOUNCED_LIGHTING
-        if conf.prop_no_shadows:
-            flags |= StaticPropFlags.NO_SHADOW
         if conf.prop_no_vert_light:
             flags |= StaticPropFlags.NO_PER_VERTEX_LIGHTING
         if conf.prop_no_self_shadow:
             flags |= StaticPropFlags.NO_SELF_SHADOWING
+        if conf.prop_no_shadows:
+            flags |= StaticPropFlags.NO_SHADOW
 
-        leafs = compute_visleafs(coll_data, ctx.bsp.vis_tree())
-        ctx.bsp.props.append(StaticProp(
-            model=model_name,
-            origin=center,
-            angles=Angle(0, 270, 0),
-            scaling=1.0,
-            visleafs=leafs,
-            solidity=6 if has_coll else 0,
-            flags=flags,
-            tint=Vec(conf.prop_rendercolor),
-            renderfx=conf.prop_renderalpha,
-            lighting=center + light_origin,
-            min_fade=conf.prop_fade_min_dist,
-            max_fade=conf.prop_fade_max_dist,
-            fade_scale=conf.prop_fade_scale,
-        ))
-        for seg_prop in seg_props:
+        for m in modellist:
+            new_flags = flags | m.flags  # For the glass, we force shadows off.
```

Comment:
TODO: Check if opaque value is 0 before forcing shadow off. Or just force combined model on opaque?